go/vt/vtgate/planbuilder/operators/ast2op.go

```diff
@@ -186,6 +359,63 @@ func createOperatorFromUpdate(ctx *plancontext.PlanningContext, updStmt *sqlpars
 	return subq, nil
 }
 
+// fkNeedsHandlingForUpdates returns what all foreign keys need handling in Vitess for the given set of update expressions.
+// Here we return all the parent and child foreign key constraints that could require any handling.
+func fkNeedsHandlingForUpdates(updateExprs sqlparser.UpdateExprs, parentFks []vindexes.ParentFKInfo, childFks []vindexes.ChildFKInfo) ([]vindexes.ParentFKInfo, []vindexes.ChildFKInfo) {
```

Comment:
this method name is weird, imho...
I would suggest that it contains the verb `filter`, since that is what it is doing, right?

Reply:
to me filter means removing if something matches a given rule.
How about renaming this to `getFKRequirementsForUpdate`